src/stream/stream/peekable.rs

```diff
@@ -0,0 +1,65 @@
+use std::pin::Pin;
+
+use crate::future::Future;
+use crate::stream::Stream;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
+#[allow(missing_debug_implementations)]
+pub struct Peekable<S: Stream>
+where
+    S: Sized,
+{
+    stream: S,
+    peeked: Option<Option<S::Item>>,
+}
+
+//pub struct PeekFuture<'a, T: Unpin + ?Sized> {
+//pub(crate) stream: &'a T,
+//}
+
+//impl<T: Stream + Unpin + ?Sized> Future for PeekFuture<'_, T> {
+//}
+
+impl<S> Peekable<S>
+where
+    S: Stream,
+{
+    pin_utils::unsafe_pinned!(stream: S);
+    pin_utils::unsafe_unpinned!(peeked: Option<Option<S::Item>>);
+
+    pub(crate) fn new(stream: S) -> Self {
+        Peekable {
+            stream: stream,
+            peeked: None,
+        }
+    }
+
+    pub fn peek(&mut self) -> impl Future<Output = Option<S::Item>> {
+        async { None }
+    }
+}
+
+impl<S> Stream for Peekable<S>
+where
+    S: Stream,
+{
+    type Item = S::Item;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        match &self.peeked {
+            Some(_) => {
+                let v = Poll::Ready(self.as_mut().peeked().take().unwrap());
+                *self.as_mut().peeked() = None;
+                v
```

Comment:
In my opinion, the `peeked` just should hold the `next`. And every `poll_next`, it will be changed to new `next`. Why do ` *self.as_mut().peeked() = None;`